src/suffix.c

```diff
@@ -124,55 +128,211 @@ void deleteSuffixTrie(Trie *trie, const char *str, uint32_t len) {
   runeBufFree(&buf);
 }
 
-static int processSuffixData(suffixData *data, TrieSuffixCallback callback, void *ctx) {
+static int processSuffixData(suffixData *data, SuffixCtx *sufCtx) {
+  //TrieSuffixCallback callback, void *ctx) {
   if (!data) {
     return REDISMODULE_OK;
   }
   arrayof(char *) array = data->array;
   for (int i = 0; i < array_len(array); ++i) {
-    if (callback(array[i], strlen(array[i]), ctx) != REDISMODULE_OK) {
+    if (sufCtx->callback(array[i], strlen(array[i]), sufCtx->cbCtx, NULL) != REDISMODULE_OK) {
       return REDISEARCH_ERR;
     }
   }
   return REDISMODULE_OK;
 }
 
-static int recursiveAdd(TrieNode *node, TrieSuffixCallback callback, void *ctx) {
+static int recursiveAdd(TrieNode *node, SuffixCtx *sufCtx) {
   if (node->payload) {
     size_t rlen;
-    // printf("nodestr %s len %d rlen %ld", runesToStr(node->str, node->len, &rlen), node->len, rlen);
     suffixData *data = Suffix_GetData(node);
-    processSuffixData(data, callback, ctx);
```

Comment:
Why we ignored the error before? Was it a bug?

Reply:
Yes. The failure can be caused by getting over `maxPrefixExpansions`.

Reply:
Did we mentioned it on the top comment?